app/Http/Controllers/App/PostController.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace App\Http\Controllers\App;
+
+use App\Http\Controllers\Controller;
+use App\Models\Banner;
+use App\Models\Challenge;
+use App\Models\Comment;
+use App\Models\Post;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Storage;
+use Inertia\Inertia;
+
+class PostController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Inertia\Response
+     */
+    public function index(Request $request)
+    {
+        $posts = Post::with(['author', 'team', 'banner', 'challenge', 'comments', 'comments.author', 'users'])->orderByDesc('updated_at')->get();
+        $posts = $posts->each(function($post){$post->team->image = ($post->team->image) ? Storage::disk('upload')->url($post->team->image) : null;});
+
+        $user = $request->user();
+        $teams = $user->teams;
+        $teams->load('banner');
+
+        return Inertia::render('post/Index', [
+            'posts' => $posts,
+            'banners' => Banner::all()->keyBy('id'),
+            'challenges' => Challenge::all(),
+            'teams' => $teams,
+        ]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $data = $this->validate($request, [
+            'subject' => ['required', 'string'],
+            'content' => ['required', 'string'],
+            'team_id' => ['required', 'exists:teams,id'],
+            'banner_id' => ['nullable', 'requiredIf:banner_related,true', 'exists:banners,id'],
```

Comment:
If this is `nullable` we don't really care about the `banner_related` field, no?
```suggestion
            'banner_id' => ['nullable', 'exists:banners,id'],
```

Reply:
I removed it